src/aatoolbox/datasources/codab/codab.py

```diff
@@ -49,50 +62,110 @@ def download(
 
         Examples
         --------
-        >>> from aatoolbox.datasources.codab.codab import CodAB
+        >>> from aatoolbox import create_country_config, CodAB
         >>> # Download COD administrative boundaries for Nepal
-        >>> codab = CodAB("npl")
-        >>> npl_cod_shapefile = codab.download(
-        ...    hdx_address="administrative-bounadries-of-nepal",
-        ...    hdx_dataset_name="npl_admbnda_nd_20201117_SHP.zip")
+        >>> country_config = create_country_config(iso3="npl")
+        >>> codab = CodAB(country_config=country_config)
+        >>> npl_cod_shapefile = codab.download()
         """
-        return self._download(
+        return _download(
             filepath=self._raw_filepath,
-            hdx_address=hdx_address,
-            hdx_dataset_name=hdx_dataset_name,
+            hdx_address=self._country_config.codab.hdx_address,
+            hdx_dataset_name=self._country_config.codab.hdx_dataset_name,
             clobber=clobber,
         )
 
-    @staticmethod
-    @check_file_existence
-    def _download(
-        filepath: Path, hdx_address: str, hdx_dataset_name: str, clobber: bool
-    ):
-        return load_dataset_from_hdx(
-            hdx_address=hdx_address,
-            hdx_dataset_name=hdx_dataset_name,
-            output_filepath=filepath,
+    def load(self, admin_level: int) -> gpd.GeoDataFrame:
+        """
+        Get the COD AB data by admin level.
+
+        Parameters
+        ----------
+        admin_level: int
+            The administrative level
+
+        Returns
+        -------
+        COD AB geodataframe with specified admin level
+
```

Comment:
Consider adding a `Raises` section to the documentation explaining which exception it might raise and why.

https://numpydoc.readthedocs.io/en/latest/format.html#raises

Reply:
Great idea, done!